### ISSUE TITLE
[java] EmptyCatchBlock false negative when allowCommentedBlocks=true

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4928]
**Affects PMD Version:** 7.0.0

**Rule:** [EmptyCatchBlock](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#emptycatchblock)

**Description:**

As commented on https://github.com/pmd/pmd/pull/4871#pullrequestreview-1980157089 - for Apex, formal (doc) comments are dealt the same way like other comments now. No special treatment.

I thought, EmptyCatchBlock does the same, since there was a test case explicitly for javadoc comments saying "Javadoc comment is not OK"... but a test case with allowCommentedBlock was missing for javadoc comments.

So it turns out, that for Java, javadoc comments are not treated specially as well. So the Apex and Java rule are similar now.

But testing further showed, it's a bit worse: When the rule is configured with `allowCommentedBlocks=true`, then empty catch blocks are not found anymore...

**Code Sample demonstrating the issue:**

```java
public class Foo {
    void bar() {
        try {} catch (Exception e) {}
    }
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.
